chromium_src/chrome/renderer/url_loader_throttle_provider_impl.h

```diff
@@ -0,0 +1,19 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_CHROMIUM_SRC_CHROME_RENDERER_URL_LOADER_THROTTLE_PROVIDER_IMPL_H_
+#define BRAVE_CHROMIUM_SRC_CHROME_RENDERER_URL_LOADER_THROTTLE_PROVIDER_IMPL_H_
+
+#define BRAVE_URL_LOADER_THROTTLE_PROVIDER_IMPL_H_                   \
```

Comment:
https://bravesoftware.slack.com/archives/CU5UXM3EC/p1705545054988689?thread_ts=1705420692.572819&cid=CU5UXM3EC

Reply:
yep we should override `URLLoaderThrottleProviderImpl::Create` so we don't have to copy 
```
#if BUILDFLAG(ENABLE_EXTENSIONS)
  mojo::PendingRemote<safe_browsing::mojom::ExtensionWebRequestReporter>
      pending_extension_web_request_reporter;
  broker->GetInterface(
      pending_extension_web_request_reporter.InitWithNewPipeAndPassReceiver());
#endif  // BUILDFLAG(ENABLE_EXTENSIONS)
```
if we keep `BraveURLLoaderThrottleProviderImpl::Create`, stuffs can be easily broken during future chromium upgrades and hard to detect.

Reply:
With that approach, we also need to override `URLLoaderThrottleProviderImpl::CreateThrottles` in order to insert our `tor::OnionDomainThrottle`

Reply:
OK, makes sense. Since we elminated the patch and given the current timing I'm going to open a follow-up issue to address this one.

Reply:
https://github.com/brave/brave-browser/issues/35897